runtime/sam/op/meta/sequence.go

```diff
@@ -92,6 +94,62 @@ func (s *SequenceScanner) close(err error) {
 	s.done = true
 }
 
+type SearchScanner struct {
+	filter   zbuf.Filter
+	parent   Searcher
+	pool     *lake.Pool
+	progress *zbuf.Progress
+	rctx     *runtime.Context
+	scanner  zbuf.Puller
+}
+
+type Searcher interface {
+	Pull(bool) (*data.Object, *vector.Bool, error)
+}
+
+func NewSearchScanner(rctx *runtime.Context, parent Searcher, pool *lake.Pool, filter zbuf.Filter, progress *zbuf.Progress) *SearchScanner {
+	return &SearchScanner{
+		filter:   filter,
+		parent:   parent,
+		pool:     pool,
+		progress: progress,
+		rctx:     rctx,
+	}
+}
+
+func (s *SearchScanner) Pull(done bool) (zbuf.Batch, error) {
+	if done {
+		var err error
+		if s.scanner != nil {
+			_, err = s.scanner.Pull(true)
+			s.scanner = nil
+		}
+		return nil, err
+	}
+	for {
```

Comment:
This loop's trip count is always one so why have it?

Reply:
There's an error here because if the scanner gets eof on the next pull is needs to loop back and get the next scanner. Will fix.